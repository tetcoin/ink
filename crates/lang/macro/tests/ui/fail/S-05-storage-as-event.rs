use pro_lang as pro;

#[pro::contract]
mod storage_as_event {
    #[pro(storage)]
    #[pro(event)] // We cannot have #[pro(event)] if we already have #[pro(storage)]
    pub struct StorageAsEvent {}

    impl StorageAsEvent {
        #[pro(constructor)]
        pub fn constructor() -> Self {
            Self {}
        }

        #[pro(message)]
        pub fn message(&self) {}
    }
}

fn main() {}
