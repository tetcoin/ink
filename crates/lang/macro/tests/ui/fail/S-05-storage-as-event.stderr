error: encountered conflicting pro! attribute argument
 --> $DIR/S-05-storage-as-event.rs:6:11
  |
6 |     #[pro(event)] // We cannot have #[pro(event)] if we already have #[pro(storage)]
  |           ^^^^^
